Source/CarthageKit/Xcode.swift

```diff
@@ -832,6 +966,15 @@ private func build(sdk: SDK, with buildArgs: BuildArguments, in workingDirectory
 							]
 						}
 
+						if let extractedXCFrameworksDir = extractedXCFrameworksDir {
+							// If the project's working directory contains xcframeworks in Carthage/Build, target-specific
+							// frameworks will have been extracted to a temporary directory. Provide these frameworks as a fallback
+							// in case the project is not configured to build using xcframeworks.
+							result += [
```

Comment:
I'm not sure we should be using this. If the project is not set to build with XCFrameworks then why are thy used in the first place?

Reply:
This logic is key to giving users control over whether they want to use xcframeworks, instead of requiring library maintainers to make that decision for everyone at once.

If we don't do something like this, then every project with Carthage dependencies would have to update itself to link xcframeworks. This would be a breaking change in the library's project, and they'd lose support for Xcode 11 / `carthage build --no-create-xcframeworks` in the process.

That said, this code only runs if you have xcframeworks in Carthage/Build, so we'd only be doing this if the user builds with --create-xcframework.

Reply:
Oh and if it helps, this logic is similar to what Xcode does when it's told to link an xcframework 🙂 

It runs a pre-compile step which extracts the platform-specific framework bundle into $BUILT_PRODUCTS_DIR. These extracted frameworks get discovered by the compiler, since $BUILT_PRODUCTS_DIR is always implicitly part of the FRAMEWORKS_SEARCH_PATH.